src/Orchard/Environment/Extensions/Folders/IExtensionFolders.cs

```diff
@@ -4,5 +4,6 @@
 namespace Orchard.Environment.Extensions.Folders {
     public interface IExtensionFolders {
         IEnumerable<ExtensionDescriptor> AvailableExtensions();
+        IEnumerable<string> Paths { get; }

```

Comment:
Make sense. 

At the moment the only location path are actually stored at is IExtensionFolders (everything else is hard coded).
I am not that familiar with Orchard classes so I'll appreciate if you could suggest a class for storing the path lists.
For the sake of efficiency I would store just path lists and not anything more fancy.